### DISCUSSION TITLE
Case raw queries support multi-word PascalCase model names?

### DISCUSSION THREAD

**Discussion by aeddie-zapidhire** [prisma/prisma#12403] [Q&A] [answered]
I'm writing a raw query like this, using the PascalCase for the model name:

```
const [result] = await this.db.$queryRaw<[{ count: number }]>`
    SELECT COUNT(*)
    FROM QuestionStage AS a
    ${getJoins(args)}
    ${getWhere(args)}
   `
```

The model is defined as:

```
model QuestionStage {
  id          String             @id @default(dbgenerated("uuid_generate_v4()")) @db.Uuid
  job_id      String             @db.Uuid
  title       String
  description String?
  ordering    Int                @default(0)
  created_at  DateTime           @default(now()) @db.Timestamptz
  updated_at  DateTime           @updatedAt @db.Timestamptz
  //
  job         Job?               @relation(fields: [job_id], references: [id], onDelete: Cascade)
  questions   QuestionStageMap[]

  @@map(name: "question_stage")
}
```

But I get the following error when funning my integration tests:

```
Raw query failed. Code: `42P01`. Message: `relation "questionstage" does not exist`
```

If I replace `QuestionStage` with `question_stage` in the query (that is, the real table name), it all works.

Am I doing something wrong or does Prisma object to a multiple upper-case letters in the model name?

Thanks in advance.

**Comment by nurul3101** [comment, marked as answer]
Hey @aeddie-zapidhire 👋 

In case of [raw database access](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access) i.e. queryRaw which you are invoking, prisma passes the query directly to the database, so you need to pass the table name which exists in the database and not the model name which exists in prisma schema.

In this case table name is `question_stage` which you have defined through `@@map` attribute, so passing `question_stage` works.

You will be using model names in all other queries except these raw database access queries.

**Reply by aeddie-zapidhire** [reply]
@nurul3101 ah ok, I misunderstood the examples in the docs (because my cognitive bias is that I would never start the name of a database table with a capital). Thank you.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
